README.md

```diff
@@ -137,16 +137,18 @@ const xmtp = await Client.create(wallet)
+| skipContactPublishing     | `false`                                                                           | Do not publish the user's contact bundle to the network on Client creation. Designed to be used in cases where the Client session is short-lived (for example, decrypting a push notification), and where it is known that a Client instance has been instantiated with this flag set to false at some point in the past.                                                                                                        |
+| codecs                    | `[TextCodec]`                                                                     | Add codecs to support additional content types.                                                                                                                                                                                                                                                                                                                                                                                  |
+| maxContentSize            | `100M`                                                                            | Maximum message content size in bytes.                                                                                                                                                                                                                                                                                                                                                                                           |
+| appVersion                | `undefined`                                                                       | Add a client app version identifier that's included with API requests. For example, you can use the following format: `appVersion: APP_NAME + '/' + APP_VERSION`. Setting this value provides telemetry that shows which apps are using the XMTP client SDK. This information can help XMTP developers provide app support, especially around communicating important SDK updates, including deprecations and required upgrades. |
+| preCreateIdentityCallback | `undefined`                                                                       | `preCreateIdentityCallback` is a function that will be called immediately before a "Create Identity" wallet signature is requested from the user.                                                                                                                                                                                                                                                                                |
```

Comment:
It's almost like this `callback` is used to "allow XMTP to pop a modal". Like a "showCreateIdentitySignatureModalCallback" or are the semantics slightly different?

Reply:
Opening a modal is one possible use of this, but I want to be agnostic to how the developer might use the feature. 

Maybe it's a toast message, maybe it's an analytics metric

Reply:
I see, I was incorrectly conceptualizing as developer calls `callback` which initiates `xmtp-js` wallet signing modal popup. But the more accurate formulation is `xmtp-js` initiates by calling the `callback`, providing the developer a chance to do whatever before `xmtp-js` prompts for signature.

Reply:
I should really include an example in the README of how to use this.

Going to do another pass at documentation before release tomorrow and will include in that.